src/django_components/dependencies.py

```diff
@@ -699,30 +711,22 @@ def _gen_exec_script(
     # NOTE: It would be better to pass only the URL itself for `loadJs/loadCss`, instead of a whole tag.
     #    But because we allow users to specify the Media class, and thus users can
     #    configure how the `<link>` or `<script>` tags are rendered, we need pass the whole tag.
-    escaped_to_load_js_tags = [_escape_js(tag, eval=False) for tag in to_load_js_tags]
-    escaped_to_load_css_tags = [_escape_js(tag, eval=False) for tag in to_load_css_tags]
-
-    # Make JS array whose items are interpreted as JS statements (e.g. functions)
-    def js_arr(lst: List) -> str:
-        return "[" + ", ".join(lst) + "]"
-
-    # NOTE: Wrap the body in self-executing function to avoid polluting the global scope.
-    exec_script: types.js = f"""
-        (() => {{
-            Components.manager._loadComponentScripts({{
-                loadedCssUrls: {json.dumps(loaded_css_urls)},
-                loadedJsUrls: {json.dumps(loaded_js_urls)},
-                toLoadCssTags: {js_arr(escaped_to_load_css_tags)},
-                toLoadJsTags: {js_arr(escaped_to_load_js_tags)},
-            }});
-            document.currentScript.remove();
-        }})();
-    """
-
-    # NOTE: The exec script MUST be executed SYNC, so we MUST NOT put `type="module"`,
-    #       `async`, nor `defer` on it.
-    #       See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#notes
-    exec_script = f"<script>{_escape_js(dedent(exec_script))}</script>"
+    escaped_to_load_js_tags = [_escape_js(tag, wrap=False) for tag in to_load_js_tags]
+    escaped_to_load_css_tags = [_escape_js(tag, wrap=False) for tag in to_load_css_tags]
+
+    exec_script_data = {
+        "loadedCssUrls": loaded_css_urls,
+        "loadedJsUrls": loaded_js_urls,
+        "toLoadCssTags": escaped_to_load_css_tags,
+        "toLoadJsTags": escaped_to_load_js_tags,
+    }
+
+    # NOTE: This data is embedded into the HTML as JSON. It is the responsibility of
+    # the client-side code to detect that this script was inserted, and to load the
+    # corresponding assets
+    # See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#embedding_data_in_html
+    exec_script = json.dumps(exec_script_data)
+    exec_script = f'<script type="application/json" data-djc>{exec_script}</script>'
```

Comment:
Previously, when we rendered a component, we inserted a `<script>` tag that contained logic to load the JS / CSS via the client-side dependency manager.

But there was a problem that browsers execute the `<script>` tags only when certain methods are used. And e.g. the scripts were not being triggered when for example when the script was inserted via `el.innerHTML`.

So I changed the approach. Instead of inserting a `<script>` tag with an executable JS, now we insert only a JSON data as `<script type="application/json">`.

And instead, our dependency manager JS now listens for changes to the DOM. And when the changes include inserting `<script type="application/json">` with the `data-djc` attribute, we know that this is coming from us. At that point we parse the JSON, and run the same logic that we originally wanted to run in the script.

This is kinda similar to AlpineJS, where they also listen for DOM changes to know if any of the Alpine attributes (e.g. `x-data`) were added or removed.